feat: Added kind parameter to getPricesParameters and getProofsParameters

chetanr25 opened the pull request:
### What
Added `Kind` parameter in getProofsParameters.
We will be needing this in order to filter out Prices data based on the kind of contribution ("Community" or "Consumption") 

With the help of kind parameter, it will be possible for us to filter the contributions based on type of contribution and can replicate the dashboard to our mobile app https://prices.openfoodfacts.org/dashboard

- Not tied to a specific issue, but essential for enabling kind-based filtering on mobile to match web dashboard behavior.

monsieurtanuki commented on lib/src/prices/get_prices_parameters.dart:
```diff
@@ -26,6 +26,7 @@ class GetPricesParameters extends GetParametersHelper<GetPricesOrderField> {
   String? owner;
   DateTime? createdGte;
   DateTime? createdLte;
+  String? kind;
```
```suggestion
  /// Typical value taken from PriceKind.offTag
  String? kind;
```

monsieurtanuki reviewed (commented):
Hi @chetanr25!

Your code seems to work. But again, it's not a good thing when there are no issue to base your PR on.
Would you please create a new file for enum `PriceKind`, similar to `LocationType`, with offTag values `COMMUNITY` and `CONSUMPTION`?

monsieurtanuki commented:
@chetanr25 Btw there's confusion between your title (about `getProofsParameters`) and your code (impacting `getPricesParameters`).
Please impact both `getProofsParameters` and `getPricesParameters`, and fix the title of your issue.

chetanr25 commented:
Thanks a lot for all the helpful hints and detailed feedback.
And again, apologise for directly creating a PR without a base issue. 

I have updated the PR title as required, created `price_kind.dart` and `proof_kind.dart` similar to `location_type.dart`, and applied the changes to both `GetPricesParameters` and `GetProofsParameters`.

Please let me know if the implementation is correct or if anything needs to be improved.

monsieurtanuki commented on lib/src/prices/get_prices_parameters.dart:
```diff
@@ -1,3 +1,5 @@
+import 'package:openfoodfacts/src/prices/price_kind.dart';
```
```suggestion
import 'price_kind.dart';
```
Please put direct path here.

monsieurtanuki commented on lib/src/prices/price_kind.dart:
```diff
@@ -0,0 +1,22 @@
+import 'package:json_annotation/json_annotation.dart';
+
```
```suggestion
```
Not needed.

monsieurtanuki commented on lib/src/prices/price_kind.dart:
```diff
@@ -0,0 +1,22 @@
+
+/// Kind of the Price.
+enum PriceKind implements OffTagged {
+  @JsonValue('CONSUMPTION')
```
```suggestion
```
Not needed.

monsieurtanuki commented on lib/src/prices/price_kind.dart:
```diff
@@ -0,0 +1,22 @@
+/// Kind of the Price.
+enum PriceKind implements OffTagged {
+  @JsonValue('CONSUMPTION')
+  online(offTag: 'CONSUMPTION'),
```
```suggestion
  consumption(offTag: 'CONSUMPTION'),
```
Let's use reasonable names, right?

monsieurtanuki commented on lib/src/prices/price_kind.dart:
```diff
@@ -0,0 +1,22 @@
+enum PriceKind implements OffTagged {
+  @JsonValue('CONSUMPTION')
+  online(offTag: 'CONSUMPTION'),
+  @JsonValue('COMMUNITY')
```
```suggestion
```
Not needed.

monsieurtanuki commented on lib/src/prices/price_kind.dart:
```diff
@@ -0,0 +1,22 @@
+  @JsonValue('CONSUMPTION')
+  online(offTag: 'CONSUMPTION'),
+  @JsonValue('COMMUNITY')
+  osm(offTag: 'COMMUNITY');
```
```suggestion
  community(offTag: 'COMMUNITY');
```

monsieurtanuki commented on lib/src/prices/proof_kind.dart:
Not needed anymore.

monsieurtanuki reviewed (commented):
Hi @chetanr25, and thank you for your work!

Please have a look at my comments.
I guess we only need a single "kind" - let's call it `ContributionKind`. Btw this file needs to be exported in `openfoodfacts.dart`.

Besides, just a little beyond the current PR, could you please make some changes in `GetParametersHelper`:
* add field `Map<String, String>? additionalParameters`
* use this field in method `getQueryParameters`

That will gives us more flexibility.

chetanr25 commented:
Thank you for the suggestion, I have commited the suggested changes. 
Could you please confirm if the implementation is correct, particularly the `additionalParameters` part.

monsieurtanuki commented on lib/src/prices/get_prices_parameters.dart:
```diff
@@ -53,6 +52,11 @@ class GetPricesParameters extends GetParametersHelper<GetPricesOrderField> {
     addNonNullDate(createdGte, 'created__gte', dayOnly: false);
     addNonNullDate(createdLte, 'created__lte', dayOnly: false);
     addNonNullString(kind?.offTag, 'kind');
+    if (additionalParameters != null) {
+      for (final String key in additionalParameters!.keys) {
+        addNonNullString(additionalParameters![key], key);
+      }
+    }
```
Please:
1. move that to the super method
2. use `_result.addAll(additionalParameters);` instead

monsieurtanuki commented on lib/src/prices/get_proofs_parameters.dart:
```diff
@@ -40,6 +39,11 @@ class GetProofsParameters
     addNonNullDate(dateLt, 'date__lt', dayOnly: true);
     addNonNullDate(dateLte, 'date__lte', dayOnly: true);
     addNonNullString(kind?.offTag, 'kind');
+    if (additionalParameters != null) {
+      for (final String key in additionalParameters!.keys) {
+        addNonNullString(additionalParameters![key], key);
+      }
+    }
```
Moved to super.

monsieurtanuki reviewed (commented):
Hi @chetanr25!
Some code to move and that's it.

chetanr25 commented:
Thank you for the suggestion, I have made the recommended changes and committed to this PR.

monsieurtanuki commented:
> Thank you for the suggestion, I have made the recommended changes and committed to this PR.

@chetanr25 You haven't completely.
Please move the `addAll` part to `GetParametersHelper`.

monsieurtanuki reviewed (approved):
Thank you @chetanr25!

chetanr25 commented:
> > Thank you for the suggestion, I have made the recommended changes and committed to this PR.
> 
> @chetanr25 You haven't completely. Please move the `addAll` part to `GetParametersHelper`.

my bad! I had added it to both classes earlier. I’ve now moved the addAll part to GetParametersHelper as suggested and committed the changes.
Thank you!
